### PR TITLE
9 Create ghcr.io/harrisonai/terraform Docker container image in Githu…

### PR DESCRIPTION
**Pull request by comozo**
Create Docker container image in Github container registry as part of a migration plan to move from Docker Hub to Github Container Registry.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+on:
+  pull_request:
+    branches:
+      - main
+  release:
+    types: [published]
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
```
Comment (dcarrion87) [inline .github/workflows/deploy.yml]: Don't think IMAGE_NAME is needed here as it's set in the makefile.. That example repository is just a sample for adaption.

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Log into ghcr
+        uses: docker/login-action@v2.1.0
+        with:
```
Comment (dcarrion87) [inline .github/workflows/deploy.yml:26]: Maybe a condition if it's a release same as deleted build.yaml?

**Review by dcarrion87 — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+name: Publish Image
+
+on:
+  pull_request:
+    branches:
+      - main
+  release:
+    types: [published]
```
Comment (dcarrion87) [inline .github/workflows/deploy.yml:8]: Feel free to adjust this to push a "test" tagged image for testing outside of the scope of a merge if you want to test it. Then switch back before merge. It's a small change so also fine to just merge and subsquent fix if there's any publishing issues.

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (comozo) [reply .github/workflows/deploy.yml:26]: Make sense

**Review by dcarrion87 — APPROVED** [review]

**Pull request merged**